examples/column/vanleer_advection.jl

```diff
@@ -0,0 +1,158 @@
+# so at time t, the solution is y(z - w * t)
+
+# visualization artifacts
+ENV["GKSwstype"] = "nul"
+using ClimaCorePlots, Plots
+Plots.GRBackend()
+dir = "vanleer_advection"
+path = joinpath(@__DIR__, "output", dir)
+mkpath(path)
+
+
+function tendency!(yₜ, y, parameters, t)
+    (; w, Δt, limiter_method) = parameters
+    FT = Spaces.undertype(axes(y.q))
+    bcvel = pulse(-π, t, z₀, zₕ, z₁)
+    divf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(FT(0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0))),
+    )
+    VanLeerMethod = Operators.LinVanLeerC2F(
+        bottom = Operators.FirstOrderOneSided(),
+        top = Operators.FirstOrderOneSided(),
+        method = limiter_method,
+    )
+
+    If = Operators.InterpolateC2F()
+
+    @. yₜ.q = -divf2c(VanLeerMethod(w, y.q, Δt))
+end
+
+# Define a pulse wave or square wave
+
+FT = Float64
+t₀ = FT(0.0)
+t₁ = FT(6)
+z₀ = FT(0.0)
+zₕ = FT(2π)
+z₁ = FT(1.0)
+speed = FT(-1.0)
+pulse(z, t, z₀, zₕ, z₁) = abs(z - speed * t) ≤ zₕ ? z₁ : z₀
+
+n = 2 .^ 8
+elemlist = 2 .^ [3, 4, 5, 6, 7, 8, 9, 10]
+Δt = FT(0.3) * (20π / n)
+@info "Timestep Δt[s]: $(Δt)"
+
+domain = Domains.IntervalDomain(
+    Geometry.ZPoint{FT}(-10π),
+    Geometry.ZPoint{FT}(10π);
+    boundary_names = (:bottom, :top),
+)
+
+stretch_fns = (Meshes.Uniform(), Meshes.ExponentialStretching(FT(7.0)))
+plot_string = ["uniform", "stretched"]
+
+for (i, stretch_fn) in enumerate(stretch_fns)
+    @info stretch_fn
+    limiter_methods = (
+        Operators.AlgebraicMean(),
+        Operators.PosDef(),
+        Operators.Mono4(),
+        Operators.Mono5(),
+    )
+    for (j, limiter_method) in enumerate(limiter_methods)
+        mesh = Meshes.IntervalMesh(domain, stretch_fn; nelems = n)
+        cent_space = Spaces.CenterFiniteDifferenceSpace(mesh)
+        face_space = Spaces.FaceFiniteDifferenceSpace(cent_space)
+        z = Fields.coordinate_field(cent_space).z
+        O = ones(FT, face_space)
+
+        # Initial condition
+        q_init = pulse.(z, 0.0, z₀, zₕ, z₁)
+        y = Fields.FieldVector(q = q_init)
+
+        # Unitary, constant advective velocity
+        w = Geometry.WVector.(speed .* O)
+
+        # Solve the ODE
+        parameters = (; w, Δt, limiter_method)
+        prob = ODEProblem(
+            ClimaODEFunction(; T_exp! = tendency!),
+            y,
+            (t₀, t₁),
+            parameters,
+        )
+        sol = solve(
+            prob,
+            ExplicitAlgorithm(SSP33ShuOsher()),
+            dt = Δt,
+            saveat = Δt,
+        )
+
+        q_final = sol.u[end].q
+
+        if limiter_method != Operators.AlgebraicMean()
+            @info "Extrema with $(limiter_method): $(extrema(q_final))"
+            @assert maximum(q_final) <= FT(1)
+            @assert isapprox(maximum(q_final .- 1), FT(0), atol = FT(0.05))
+            @assert isapprox(minimum(q_final .- 0), FT(0), atol = FT(0.05))
+        end
```

Comment:
This looks good! I'd expect a lower error than 5%, but maybe if `Δt` is big for this particular problem, and if there are several stages that each introduce a bit of error, then this result makes sense. Could you also add a similar monotonicity check for the `RZeroLimiter` in the TVD advection test, and any other limiters that should preserve monotonicity?

Reply:
Does this error decrease if you lower the value of `Δt`? It would be good to have a unit test that demonstrates monotonicity preservation up to machine epsilon precision, if that's possible for a sufficiently small CFL number in this test case.

Reply:
@dennisYatunin Updated with reduced cfl and info statements.